components/builder-worker/src/runner/mod.rs

```diff
@@ -102,20 +102,28 @@ impl Runner {
 
     pub fn run(mut self) -> Job {
         if let Some(err) = util::validate_integrations(&self.workspace).err() {
-            error!("failed to validate integrations err={:?}", err);
+            let msg = format!("Failed to validate integrations, err={:?}", err);
+            error!("{}", msg);
+            self.logger.log(&msg);
             return self.fail(net::err(ErrCode::INVALID_INTEGRATIONS, "wk:run:7"));
         };
         if let Some(err) = self.setup().err() {
-            error!("failed to setup workspace err={:?}", err);
+            let msg = format!("Failed to setup workspace, err={:?}", err);
+            error!("{}", msg);
+            self.logger.log(&msg);
             return self.fail(net::err(ErrCode::WORKSPACE_SETUP, "wk:run:1"));
         }
         if let Some(err) = self.install_origin_secret_key().err() {
-            error!("failed to retrieve secret key, err={:?}", err);
+            let msg = format!("Failed to install origin secret key, err={:?}", err);
+            error!("{}", msg);
```

Comment:
Can users put themselves in a bad state to generate this message or is always an actual error we want to be woken up about?